pkg/sdn/plugin/pod_linux.go

```diff
@@ -264,6 +268,91 @@ func vnidToString(vnid uint32) string {
 	return strconv.FormatUint(uint64(vnid), 10)
 }
 
+// podIsExited returns true if the pod is exited (all containers inside are exited).
+func podIsExited(p *kcontainer.Pod) bool {
+	for _, c := range p.Containers {
+		if c.State != kcontainer.ContainerStateExited {
+			return false
+		}
+	}
+	return true
+}
+
+// getNonExitedPods returns a list of pods that have at least one running container.
+func (m *podManager) getNonExitedPods() ([]*kcontainer.Pod, error) {
+	ret := []*kcontainer.Pod{}
+	pods, err := m.host.GetRuntime().GetPods(true)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to retrieve pods from runtime: %v", err)
+	}
+	for _, p := range pods {
+		if podIsExited(p) {
+			continue
+		}
+		ret = append(ret, p)
+	}
+	return ret, nil
+}
+
+// ipamGarbageCollection will release unused IP.
+// openshift-sdn uses the CNI host-local IPAM plugin which stores allocated IPs
+// in a file. Each file created under the IPAM directory has the format:
```

Comment:
ugh. so is this all going to go away when we get kubernetes/kubernetes#37036? If not, then should it go into the host-local plugin instead? (Particularly so that if we rebase CNI later and they've changed things, we'll get a merge failure rather than a silent breakage.)